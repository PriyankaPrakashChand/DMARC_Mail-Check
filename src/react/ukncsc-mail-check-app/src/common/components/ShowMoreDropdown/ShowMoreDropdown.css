.ShowMoreDropdown--title {
  display: inline-block;
  line-height: 1em;
}

.ShowMoreDropdown--icon {
  color: #2b88c8;
}

.ShowMoreDropdown--title-text {
  color: #2b88c8;
  text-decoration: underline;
  padding-right: 3px;
}

.ShowMoreDropdown--content {
  border-left: 5px solid #bfc2c3;
  padding: 0px 25px 0px 20px;
  margin-left: 7px;
}
