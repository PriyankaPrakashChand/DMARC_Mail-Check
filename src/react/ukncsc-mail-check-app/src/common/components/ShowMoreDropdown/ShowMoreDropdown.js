import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Accordion, Icon } from 'semantic-ui-react';

import './ShowMoreDropdown.css';

class ShowMoreDropdown extends Component {
  state = { showMore: false };

  toggleContent = () => this.setState({ showMore: !this.state.showMore });

  render() {
    const { children, title } = this.props;
    const { showMore } = this.state;

    return (
      <Accordion>
        <Accordion.Title
          active={showMore}
          onClick={this.toggleContent}
          className="ShowMoreDropdown--title"
        >
          <Icon name="dropdown" className="ShowMoreDropdown--icon" />
          <span className="ShowMoreDropdown--title-text">{title}</span>
        </Accordion.Title>
        <Accordion.Content active={showMore}>
          <div className="ShowMoreDropdown--content">{children}</div>
        </Accordion.Content>
      </Accordion>
    );
  }
}

ShowMoreDropdown.defaultProps = {
  title: 'Show more',
};

ShowMoreDropdown.propTypes = {
  title: PropTypes.string,
};

export default ShowMoreDropdown;
