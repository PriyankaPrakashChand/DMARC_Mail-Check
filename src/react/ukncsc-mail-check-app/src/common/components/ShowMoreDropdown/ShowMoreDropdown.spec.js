import React from 'react';
import { render } from 'react-testing-library';
import ShowMoreDropdown from './ShowMoreDropdown';

describe('ShowMoreDropdown', () => {
  let container;

  beforeEach(() => {
    ({ container } = render(
      <ShowMoreDropdown>
        <p>some extra content!</p>
      </ShowMoreDropdown>
    ));
  });

  test('it should match the snapshot', () =>
    expect(container).toMatchSnapshot());
});
