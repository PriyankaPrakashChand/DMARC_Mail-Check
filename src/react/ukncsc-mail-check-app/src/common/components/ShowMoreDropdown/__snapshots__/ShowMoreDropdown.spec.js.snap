// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMoreDropdown it should match the snapshot 1`] = `
<div>
  <div
    class="accordion ui"
  >
    <div
      class="title ShowMoreDropdown--title"
    >
      <i
        aria-hidden="true"
        class="dropdown icon ShowMoreDropdown--icon"
      />
      <span
        class="ShowMoreDropdown--title-text"
      >
        Show more
      </span>
    </div>
    <div
      class="content"
    >
      <div
        class="ShowMoreDropdown--content"
      >
        <p>
          some extra content!
        </p>
      </div>
    </div>
  </div>
</div>
`;
